src/Domain/CoordinateMaps/TimeDependent/Translation.cpp

```diff
@@ -82,56 +113,186 @@ std::array<tt::remove_cvref_wrap_t<T>, Dim> Translation<Dim>::frame_velocity(
+    return result;
+  }
+}
+template <size_t Dim>
+template <typename T>
+tnsr::Ij<tt::remove_cvref_wrap_t<T>, Dim, Frame::NoFrame>
+Translation<Dim>::inv_jacobian(
+    const std::array<T, Dim>& source_coords, const double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) const {
+  if (f_of_r_ == nullptr) {
+    return identity<Dim>(dereference_wrapper(source_coords[0]));
+  } else {
+    return determinant_and_inverse(
+               jacobian(source_coords, time, functions_of_time))
+        .second;
+  }
+}
+
+template <size_t Dim>
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, Dim> Translation<Dim>::coord_helper(
+    const std::array<T, Dim>& source_coords, double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time,
+    const size_t function_or_deriv_index) const {
   ASSERT(functions_of_time.count(f_of_t_name_) == 1,
          "The function of time '" << f_of_t_name_
                                   << "' is not one of the known functions of "
                                      "time. The known functions of time are: "
                                   << keys_of(functions_of_time));
-  std::array<tt::remove_cvref_wrap_t<T>, Dim> result{};
-  const auto function_of_time_and_deriv =
-      functions_of_time.at(f_of_t_name_)->func_and_deriv(time);
-  const DataVector& velocity = function_of_time_and_deriv[1];
-  ASSERT(velocity.size() == Dim,
+  const auto func_or_deriv_of_time =
+      functions_of_time.at(f_of_t_name_)
+          ->func_and_deriv(time)[function_or_deriv_index];
+  ASSERT(func_or_deriv_of_time.size() == Dim,
          "The dimension of the function of time ("
-             << velocity.size()
+             << func_or_deriv_of_time.size()
              << ") does not match the dimension of the translation map (" << Dim
              << ").");
+  std::array<tt::remove_cvref_wrap_t<T>, Dim> result{};
   for (size_t i = 0; i < Dim; i++) {
-    gsl::at(result, i) = make_with_value<tt::remove_cvref_wrap_t<T>>(
-        dereference_wrapper(gsl::at(source_coords, i)), velocity[i]);
+    gsl::at(result, i) = gsl::at(source_coords, i);
+  }
+  auto radial_function_value = make_with_value<tt::remove_cvref_wrap_t<T>>(
+      dereference_wrapper(source_coords[0]), 1.0);
+  if (f_of_r_ != nullptr) {
+    // finding the radius
+    radial_function_value = (*f_of_r_)(magnitude(result - center_));
+  }
+  // The option 0. specifies using the function_of_time and anything else
+  // specifies using the derivative of the function_of_time
+  if (function_or_deriv_index == 0) {
+    for (size_t i = 0; i < Dim; i++) {
+      gsl::at(result, i) += func_or_deriv_of_time[i] * radial_function_value;
+    }
+  } else {
+    for (size_t i = 0; i < Dim; i++) {
+      gsl::at(result, i) = func_or_deriv_of_time[i] * radial_function_value;
+    }
```

Comment:
...then just always `+=` here. No need for the `if` check. (And remove the comment, I think that was for a previous version)